juju/application.py

```diff
@@ -312,23 +312,40 @@ def collect_metrics(self):
             [],
         )
 
+    async def get_annotations(self):
+        """Get annotations on this application.
+
+        :return dict: The annotations for this application
+        """
+
+        facade = client.AnnotationsFacade.from_connection(
+            self.connection)
+
+        result = (await facade.Get([{"tag": self.tag}])).results[0]
+        if result.error is not None:
+            raise JujuError(result.error)
+
+        return result.annotations
+
     async def set_annotations(self, annotations):
         """Set annotations on this application.
 
         :param annotations map[string]string: the annotations as key/value
             pairs.
 
         """
+        # TODO: ensure annotations is dict with only string keys
+        # and values.
         log.debug('Updating annotations on application %s', self.name)
 
-        self.ann_facade = client.AnnotationsFacade.from_connection(
+        facade = client.AnnotationsFacade.from_connection(
             self.connection)
 
         ann = client.EntityAnnotations(
             entity=self.tag,
             annotations=annotations,
         )
-        return await self.ann_facade.Set([ann])
+        return await facade.Set([ann])
```

Comment:
Do we actually want to return anything here?  And do we need to check the result object for any errors?

Reply:
Hmm... probably. Looked for examples in the code and all the examples I found for set methods were using this pattern.  Something to queue up for later maybe?

Reply:
+1 to remaining consistent.  The return value seems unnecessary and a bit leaky, but isn't really an issue, so maybe refactor at some point down the line, yeah.